src/ui/public/index_patterns/index_patterns.js

```diff
@@ -41,6 +41,7 @@ export function IndexPatternsProvider(Notifier, Private) {
   self.cache = patternCache;
   self.getIds = getProvider('id');
   self.getTitles = getProvider('attributes.title');
+  self.getIndexPatterns = getProvider();
```

Comment:
I mean I don't remember what it means, I'd guess it just means "the concerns of this module are different from the indexPattern cache". Maybe I was thinking I might stick the ids in the indexPattern cache with the `$ids` key or something and didn't like the possibility of it colliding... not sure...